tools/chapel-py/chapel.cpp

```diff
@@ -425,6 +425,18 @@ static PyObject* AstNodeObject_attribute_group(AstNodeObject *self, PyObject *Py
                      self->astNode->attributeGroup());
 }
 
+static PyObject* AstNodeObject_pragmas(AstNodeObject *self, PyObject *Py_UNUSED(ignored)) {
+  PyObject* elms = PySet_New(NULL);
+  auto attrs = self->astNode->attributeGroup();
+  if (attrs) {
+    for (auto p: attrs->pragmas()) {
+      PyObject* s = Py_BuildValue("s", chpl::uast::pragmatags::pragmaTagToName(p));
+      PySet_Add(elms, s);
```

Comment:
Not sure myself, but do you need a DECREF here, or does `Add` "consume" the argument?